warp10/src/main/java/io/warp10/script/filler/FillerNext.java

```diff
@@ -62,4 +65,31 @@ public int getPostWindow() {
   public int getPreWindow() {
     return 0;
   }
+
+  @Override
+  public WarpScriptSingleValueFillerFunction compute(GeoTimeSerie gts) throws WarpScriptException {
+
+    // fill always sort input before.
```

Comment:
You cannot assume a single call path for a public method, so call gts.sort() which will return immediately if the GTS is already sorted.